jobs/build-podman-os.Jenkinsfile

```diff
@@ -0,0 +1,347 @@
+               }
+           }]}
+        }
+        stage("Compress Artifacts") {
+            parallel archinfo.keySet().collectEntries{arch -> [arch, {
+                pipeutils.withOptionalExistingCosaRemoteSession(
+                arch: arch, session: archinfo[arch]['session']) {
+                    parallel artifacts.collectEntries{artifact -> [artifact["platform"] , {
+                        def file = "podman-$arch-${shortcommit}.${artifact["suffix"]}"
+                        shwrap("""cosa shell -- /usr/bin/zstd -14 $file""")
+                    }]}
+                }
+            }]}
+        }
+        stage("Sync Artifacts") {
+            parallel archinfo.keySet().collectEntries{arch -> [arch, {
+                   pipeutils.withOptionalExistingCosaRemoteSession(
+                   arch: arch, session: archinfo[arch]['session']) {
+                       if (arch != "x86_64") {
+                           parallel artifacts.collectEntries{artifact -> [artifact["platform"] , {
+                               shwrap("""cosa remote-session sync {:,}podman-$arch-${shortcommit}.${artifact["suffix"]}.zst """)
+                           }]}
+                       }
+                   }
+            }]}
+        }
+        withCredentials([file(credentialsId: 'ravanelli-push-registry-secret', variable: 'REGISTRY_SECRET')]) {
+            stage("Push Manifest") {
+
+                def manifest = "podman-v5"
+                def cmds = ""
+                parallel archinfo.keySet().collectEntries{arch -> [arch, {
+                    artifacts.each { artifact ->
+                        def file = "podman-$arch-${shortcommit}.${artifact["suffix"]}.zst"
+                        def platform = "${artifact["platform"]}"
+                        cmds += " && /usr/bin/buildah manifest add --artifact --artifact-type='' \
+                                --os=linux --arch=${arch} --annotation 'disktype=${platform}' ${manifest} ${file}"
+                    }
+                }]}
+                //  Supermin is unable to access the secret mounted in another directory.
+                // Although this solution is not ideal, let's proceed with it until we
+                // can find a more effective approach.
+                // This doc includes a 'use' case that didn't function as expected.
+                // https://www.jenkins.io/doc/pipeline/steps/credentials-binding/
+                shwrap("""cp ${REGISTRY_SECRET} push-secret""")
+
+                // We need to run all buildah commands in the same superminvm to not lose
+                // the manifest creation/additions
+                shwrap("""cosa supermin-run bash -c \
+                    "/usr/bin/buildah manifest create ${manifest} ${cmds} && \
+                    /usr/bin/buildah manifest push --authfile ./push-secret --all ${manifest} \
+                    docker://${params.CONTAINER_REGISTRY_STAGING_REPO}:5.0-${shortcommit}"
+                """)
```

Comment:
in the future we should look into trying to do this outside of supermin.

right now in FCOS we don't use supermin when combining into a manifest list and pushing to the registry:

https://github.com/coreos/fedora-coreos-pipeline/blob/6eebb70044dc67a349e826006ebcbf922e8b5b24/jobs/release.Jenkinsfile#L267-L271

which just uses https://github.com/coreos/coreos-assembler/blob/main/src/cmd-push-container-manifest


we also don't use it when building cosa:

https://github.com/coreos/fedora-coreos-pipeline/blob/6eebb70044dc67a349e826006ebcbf922e8b5b24/jobs/build-cosa.Jenkinsfile#L131-L154


Maybe the `export STORAGE_DRIVER=vfs` is the key.

Reply:
The `build-cosa` doesn't use kvm. Since we need supermin to run osbuild we need to enable it. Seems it is causing issues with podman. For `release` the  `push-container-manifest` uses skopeo

Reply:
> Since we need supermin to run osbuild we need to enable it.

right. we need supermin for when we run `osbuild`, but we're not running `osbuild` here, we're running `buildah`. 



> For `release` the `push-container-manifest` uses skopeo

right. `podman`, `buildah`, `skopeo` all use the same libraries underneath. I suspect we can figure out how to run `buildah manifest` commands here without `supermin` too, but we don't need to block on that.